Amplify/Core/Support/AmplifyAsyncSequence.swift

```diff
@@ -1,11 +1,21 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
 import Foundation
 
-public struct AmplifySequence<Element: Sendable>: AsyncSequence {
+public struct AmplifyAsyncSequence<Element: Sendable>: AsyncSequence, Cancellable {
     public typealias Iterator = AsyncStream<Element>.Iterator
     private var asyncStream: AsyncStream<Element>! = nil
     private var continuation: AsyncStream<Element>.Continuation! = nil
+    private var parent: Cancellable? = nil
 
-    public init(bufferingPolicy: AsyncStream<Element>.Continuation.BufferingPolicy = .unbounded) {
+    public init(parent: Cancellable? = nil,
```

Comment:
```swift
actor ChildTask<InProcess, Success, Failure: Error>: BufferingSequence {
    typealias Element = InProcess
    let parent: Cancellable
    var inProcessChannel: AmplifyAsyncSequence<InProcess>? = nil
    var valueContinuations: [CheckedContinuation<Success, Error>] = []
    var storedResult: Result<Success, Failure>? = nil
    var isCancelled = false

    var inProcess: AmplifyAsyncSequence<InProcess> {
        let channel: AmplifyAsyncSequence<InProcess>
        if let inProcessChannel = inProcessChannel {
            channel = inProcessChannel
        } else {
            channel = AmplifyAsyncSequence<InProcess>(parent: parent, bufferingPolicy: bufferingPolicy)
            inProcessChannel = channel
        }

        // finish channel if there is already a result
        if storedResult != nil || isCancelled {
            channel.finish()
        }
        return channel
    }

    var value: Success {
        get async throws {
            try await withTaskCancellationHandler(handler: {
                Task {
                    await cancel()
                }
            }, operation: {
                try await withCheckedThrowingContinuation { continuation in
                    if isCancelled {
                        // immediately cancel is already cancelled
                        continuation.resume(throwing: CancellationError())
                    } else if let result = storedResult {
                        // immediately send result if it is available
                        valueContinuations.append(continuation)
                        send(result)
                    } else {
                        // capture contination to use later
                        valueContinuations.append(continuation)
                    }
                }
            })
        }
    }

    init(parent: Cancellable) {
        self.parent = parent
    }
```

How does storage verse API/datastore pass the parent operation from the child task to the AmplifyAsyncSequence? I think API/datastores needs it, but storage does not